packages/block-editor/src/components/provider/index.js

```diff
@@ -7,31 +7,40 @@ import { withDispatch, withSelect } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
 
 class BlockEditorProvider extends Component {
-	constructor( props ) {
-		super( ...arguments );
-		props.updateEditorSettings( props.settings );
-		props.initBlocks( props.value );
-		this.persistedValue = props.select( 'core/block-editor' ).getBlocks();
+	componentDidMount() {
+		this.props.updateEditorSettings( this.props.settings );
+		this.props.resetBlocks( this.props.value );
+
+		this.isSyncingBlockValue = true;
```

Comment:
The reason for these being in the constructor was to avoid two rerenderings and I believe it was fixing an e2e test too but maybe it's not needed anymore.

Reply:
> The reason for these being in the constructor was to avoid two rerenderings and I believe it was fixing an e2e test too but maybe it's not needed anymore.

You may be right. My intuition was toward avoiding side effects in the constructor.

https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9